### DISCUSSION TITLE
Listetning to javascript CustomEvent

### DISCUSSION THREAD

**Discussion by wr0nggg** [livewire/livewire#6820] [1. Help]
Hi, is there  a way to listen to CustomJavascript event?, im dispatching it like 
            let customEvent = new CustomEvent('custom-event')
            document.dispatchEvent(customEvent)
, inside component '#[On(custom-event)] does nothing, thanks

**Comment by joshhanley** [comment]
@wr0nggg try dispatching the event on the window instead of the document, and see if that works.

**Reply by chas688** [reply]
It may be that you need to write your JS inside a [Livewire ready](livewire:initialized) closure.. See the docs.
`<script>
    document.addEventListener('livewire:initialized', () => {
        @this.on('post-created', (event) => {
            @this.dispatch('refresh-posts'); 
        });
    });
</script>`

https://livewire.laravel.com/docs/events#dispatching-livewire-events-from-script-tags
